src/components/SearchModal/CurrencySearch/CurrencySearch.component.tsx

```diff
@@ -72,11 +72,12 @@ export const CurrencySearch = ({
   const filteredSortedTokens = useSortedTokensByQuery(sortedTokens, debouncedQuery)
 
   const filteredSortedTokensWithNativeCurrency: Currency[] = useMemo(() => {
-    if (!showNativeCurrency) return filteredSortedTokens
-    const s = debouncedQuery.toLowerCase().trim()
-    if (nativeCurrency.symbol && nativeCurrency.symbol.toLowerCase().startsWith(s)) {
+    if (!showNativeCurrency || !nativeCurrency.symbol || !nativeCurrency.name) return filteredSortedTokens
+
+    if (new RegExp(debouncedQuery.replace(/\s/g, ''), 'gi').test(`${nativeCurrency.symbol} ${nativeCurrency.name}`)) {
       return nativeCurrency ? [nativeCurrency, ...filteredSortedTokens] : filteredSortedTokens
     }
```

Comment:
```suggestion
    if (nativeCurrency  && (new RegExp(debouncedQuery.replace(/\s/g, ''), 'gi').test(`${nativeCurrency.symbol} ${nativeCurrency.name}`)) {
      return [nativeCurrency, ...filteredSortedTokens]
    }
```

Maybe in a single check :)